### ISSUE TITLE
Datastore local_db.bin is deleted between rebuilds

### ISSUE THREAD

**Issue by bes** [GoogleCloudPlatform/gradle-appengine-plugin#190]
When running the debug server locally, and using the datastore, the datastore is created in 
```
build/exploded-app/WEB-INF/appengine-generated/local_db.bin
```
Unfortunately that file is deleted between rebuilds.
There exist some workarounds:
* Save it before rebuild / copy it back
* Make a debug servlet that repopulates the datastore into a known state

However, I feel this should be taken care of by the gradle-appengine-plugin.
(And the documentation, which doesn't use this plugin, states that the datastore should be kept between debug server restarts https://cloud.google.com/appengine/docs/java/tools/devserver)

**Comment by loosebazooka** [comment]
Yeah this is a quirk to the archive generation that deletes the old archive. What you can do is set the datastore location to somewhere that doesn't get blown away on rebuilds. Add the jvm flag :

    appengine {
        jvmFlags = ['-Ddatastore.backing_store="/path/to/datastore/file/location/local_db.bin']
    }

**Comment by mwyszomierski** [comment]
This works for the datastore.bin - but datastore-indexes-auto.xml still gets generated under the /build folder, for example. I'm also trying to use the google cloud storage api. For local development, it puts its temporary files in the same /build/exploded-app/WEB-INF/appengine-generated location - which gets deleted every time. 

Is there any other way to fix this?

**Comment by ramesh-dev** [comment]
Hi mwyszomierski, 

am also into the same problem, it can be fixed by adding --generated_dir property to jvmFlags which currently not supported in this plugin, check out the issue #237

**Comment by paulkore** [comment]
Ditto here - experiencing the "appengine-generated" deletion issue on OSX.
Looking forward to the fix!

**Comment by stefanhaustein** [comment]
(I had to restart Android Studio to get the jvmFlags gradle hack working. Not sure if this is expected, but may make sense to point out)
